src/lib/db.js

```diff
@@ -905,10 +852,147 @@ export const DB = {
+    const importedCharacters = sourceCharacters.map((x) => x.id)
+    console.log('importing', importedCharacters)
+    // Add missing relics and update existing ones
+    const newRelics = Utils.clone(sourceRelics)
+    for (const relic of newRelics) {
+      const match = findRelicMatch(relic, oldRelics)
+
+      if (match) {
+        match.substats = relic.substats
+        match.main = relic.main
+        match.enhance = relic.enhance
+        match.verified = relic.verified
+        match.id = relic.id // idk why this line is needed but without it there's a chance of errors
+        if (importedCharacters.includes(relic.equippedBy)) match.equippedBy = relic.equippedBy // Only update wearer for imported character(s)
+      } else {
+        if (!importedCharacters.includes(relic.equippedBy)) relic.equippedBy = undefined // wearer doesn't necessarily exist in the save file
+        oldRelics.push(relic)
+      }
+    }
+
+    // Add missing characters and update lc/eidolon/equipped of existing ones
+    for (const character of sourceCharacters) {
+      const match = oldCharacters.find((x) => x.id == character.id)
+      const equipped = {}
+      for (const key of Object.keys(character.equipped)) {
+        equipped[key] = character.equipped[key].id
+      }
+      const form = {}
+      for (const key of Object.keys(character.form)) {
+        if (key == 'characterLevel' || key == 'lightConeLevel') {
+          form[key] = 80
+        } else {
+          form[key] = character.form[key]
+        }
+      }
+      if (match) {
+        match.form = form
+        match.equipped = equipped
+      } else {
+        oldCharacters.push({
+          equipped: equipped,
+          id: character.id,
+          form: form,
+        })
+      }
+    }
+
+    // Update relic 'equippedBy' values
+    for (const relic of oldRelics) {
+      if (!relic.equippedBy) continue
+      const wearer = oldCharacters.find((x) => x.id == relic.equippedBy)
+      if (wearer.equipped[relic.part] != relic.id) {
+        relic.equippedBy = undefined
+      }
+    }
+
+    // Update character 'equipped' values for characters not included in the partial import
+    for (const character of oldCharacters) {
+      for (const slot of Object.keys(character.equipped)) {
+        if (!character.equipped[slot]) continue
+        if (oldRelics.find((x) => x.id == character.equipped[slot]).equippedBy != character.id) {
+          character.equipped[slot] = undefined
+        }
+      }
+    }
+
+    oldRelics.map((x) => RelicAugmenter.augment(x))
+    DB.setRelics(oldRelics)
+    DB.setCharacters(oldCharacters)
+    window.refreshRelicsScore()
+
+    if (window.characterGrid?.current?.api) {
+      window.characterGrid.current.api.updateGridOptions({ rowData: oldCharacters })
+      window.characterGrid.current.api.redrawRows()
+    }
+    if (window.relicsGrid?.current?.api) {
+      window.relicsGrid.current.api.updateGridOptions({ rowData: oldRelics })
+    }
+  },
```

Comment:
I don't think we should duplicate the code here, it can probably be solved with a more targeted approach modifying mergePartialRelicsWithState